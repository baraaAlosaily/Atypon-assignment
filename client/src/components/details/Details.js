import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import { getRecipe } from '../../services/rest.cleint.service';
import './details.css';

const Details = (props) => {
    const { id } = props.match.params;
    const [recipesDetails, setRecipesDetails] = useState([]);

/*
@function that handle data that will retrived from recipe information api 
and return an object of information of recipes 
*/
    const getRecipeDetails = async () => {
        const res = await getRecipe(id)
        setRecipesDetails(res)
    }

    useEffect(() => {
        getRecipeDetails()
    }, [])

    /*
    @function that handle readmore read less feature
    */
    const ReadMore = ({ children }) => {
        const text = children;
        const [isReadMore, setIsReadMore] = useState(true);
        const toggleReadMore = () => {
          setIsReadMore(!isReadMore);
        };
        return (
          <p className="text">
            {isReadMore ? text.slice(0, 400) : text}
            <span onClick={toggleReadMore} className="read-or-hide">
              {isReadMore ? "...read more" : " show less"}
            </span>
          </p>
        );
      };

    return (
        <div>
            <div style={{"marginTop":"20px"}} className="recipes__possibility section__padding2" id="possibility">
            <div className='recipes__header-content2'>
            <h4 className="gradient__text">{recipesDetails.title}</h4>
            </div>
            </div>
       
        <div className="recipes__possibility section__padding2">
            <div className="recipes__possibility-image">
                <img src={recipesDetails.image} alt="possibility" />
            </div>
            <div className="recipes__possibility-container container">
                <p className='gradient__text'>Summary</p>
                {recipesDetails.summary&&
                 <p>
                     <ReadMore>
                     {recipesDetails.summary.replace(/<b>|<\/b>|<a>|<\/a>|<a|href=|>/gi," ")}
                     </ReadMore>
                 </p>
                 }
                <h4><a href={recipesDetails.sourceUrl}>More Information</a></h4>
            </div>
        </div>
        </div>
    )
}

export default Details

