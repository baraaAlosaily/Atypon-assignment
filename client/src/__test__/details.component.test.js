import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Details from '../components/details/Details';
Enzyme.configure({ adapter: new Adapter() })

test('Details component recieves the params successfully', () => {
  const match = { params: { id: 511728 } }
  const component = shallow(<Details match={match} />)
  expect(component).toMatchSnapshot();
});