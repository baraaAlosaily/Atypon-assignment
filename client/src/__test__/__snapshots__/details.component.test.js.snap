// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component recieves the params successfully 1`] = `ShallowWrapper {}`;

exports[`Link changes the class when hovered 1`] = `ShallowWrapper {}`;
